### DISCUSSION TITLE
Retrieve the current $lng in routes in `entry.client.tsx`

### DISCUSSION THREAD

**Discussion by binajmen** [sergiodxa/remix-i18next#89] [Q&A]
Hi,

Playing with a vanilla project, I noticed that although I hit a `fr`ench route, the english translations are loaded as well:

<img width="608" alt="Screenshot 2022-08-05 at 13 22 53" src="https://user-images.githubusercontent.com/15611419/183067552-d046e915-f6b1-4db6-a414-6879831fd72d.png">

In `entry.server`, I'm using `lng = remixContext.matches[0].params.lng ?? i18n.fallbackLng` to retrieve the language from the route. It works well as the resulting html is french translated.

But in `client.server`, the documentation suggests `.use(LanguageDetector)` which will resolve in `en` as my OS/browser language is english. How am I suppose to retrieve the language from the `routes/$lng/...` and avoid this unwanted download?

**Comment by sergiodxa** [comment]
Are you configuring the entry.client language detector to have `order: ["htmlTag"]`? By using that it will detect the language from your `<html lang>` attribute, and server-side you should set that to the language you detected.

Another option is to use `window.__remixContext` which is like he entry.server Remix context object, but I recommend you to set the `<html lang>` attribute in your document.
